axelrod/strategies/axelrod_first.py

```diff
@@ -206,6 +206,83 @@ def strategy(self, opponent: Player) -> Action:
         return random_choice(p)
 
 
+class Graaskamp(Player):
+    """
+    This strategy follows the following rules:
+
+    1. Plays Tit For Tat for the first 50 rounds;
+    2. Defects on round 51;
+    3. Plays 5 further rounds of Tit For Tat;
+    4. A check is then made to see if the opponent is playing randomly in which case
+       it defects for the rest of the game;
+    5. The strategy also checks to see if the opponent is playing Tit For Tat
+       or another strategy from a preliminary
+       tournament called ‘Analogy’ (There is no information about 'Analogy' tournament. Step 5 is not implemented fully.)
+       If so it plays Tit For Tat. If not it cooperates
+       and randomly defects every 5 to 15 moves.
+
+    This strategy came 9th in Axelrod’s original tournament.
+
+    Names:
+
+    - Graaskamp: [Axelrod1980]_
+    """
+
+    name = "Graaskamp"
+    classifier = {
+        "memory_depth": 1,
+        "stochastic": True,
+        "makes_use_of": set(),
+        "long_run_time": False,
+        "inspects_source": False,
+        "manipulates_source": False,
+        "manipulates_state": False,
+    }
+
+    def __init__(self, alpha: float = 0.05) -> None:
+        """
+        Parameters
+        ----------
+        alpha: float
+            The significant level of p-value from chi-squared test with
+            alpha == 0.05 by default.
+        """
+        super().__init__()
+        self.alpha = alpha
+        self.opponent_is_random = False
+        self.next_random_defection_turn = None
+
+    def strategy(self, opponent: Player) -> Action:
+        """This is the actual strategy"""
+        # First move
+        # Cooperate in first round as in Tit for Tat strategy
+        if not self.history:
+            return C
+        # Copy opponent's last move as in Tit for Tat strategy
+        if 52 <= len(self.history) <= 56 or len(self.history) == 51:
```

Comment:
As is this implies that the strategy will cooperate during the first 50 turns but it should not, it should play tit for tat, it should be implemented as:

```
if len(self.history) <= 56:
    if opponent.history[-1] == D or len(self.history) == 51:
        return D
    return C
```

Reply:
> Play Tit For Tat for the first 50 rounds;

Reply:
Right, I've implemented it wrongly, already fixed :+1: